pkg/mappings.bzl

```diff
@@ -72,15 +72,21 @@ strip_prefix = struct(
     from_root = _sp_from_root,
 )
 
-def pkg_attributes(mode = None, user = None, group = None, **kwargs):
+def pkg_attributes(
+        mode = None,
+        user = None,
+        group = None,
+        uid = None,
+        gid = None,
+        **kwargs):
```

Comment:
This needs some more consistency checks.  It seems that at least `user`/`uid` and `group`/`gid` should be mutually exclusive, and perhaps `uid` and `gid` require each other.

Reply:
Are they really exclusive?  IIRC, the tar format allows both.  This, of course, it prone to strangeness, like allowing   user=root, uid=42. I don't know if we should allow the footgun or not.

Reply:
At this point, I think we document it, and let folks point the foot gun.  Tar allows both, but yes, it is a footgun.